core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/DedupAggregator.java

```diff
@@ -60,7 +61,7 @@
   private DedupConfig.DedupFunctionInfo filterFunction;
   private SelectionFunction selectionFunction;
   private static final EnumSet<Schema.Type> ALLOWED_SCHEMA_TYPES = EnumSet.of(Schema.Type.INT, Schema.Type.LONG,
-          Schema.Type.FLOAT, Schema.Type.DOUBLE);
+          Schema.Type.FLOAT, Schema.Type.DOUBLE, Schema.Type.BYTES, Schema.Type.STRING);
```

Comment:
We should explicitly check if the fields are either DateTime or BigDecimal, not only checking the underlying type.

Also for BigDecimals, we need to verify if the underlying object is a BigDecimal instance, or transform it from bytes to BigDecimal using https://github.com/cdapio/cdap/blob/3a90d7c854bda94d894153f7fb6d99d51ff09ce0/cdap-api-common/src/main/java/io/cdap/cdap/api/common/Bytes.java#L801

Reply:
Agree. For BigDecimals, byets comparison would give wrong order. We need to compare specific values

Reply:
yup agree.